### ISSUE TITLE
Fails to launch on any java newer than 10

### ISSUE THREAD

**Issue by ea7ababe** [jMonkeyEngine/sdk#349]
With either `java-18-openjdk` or `java-17-openjdk`:
```
$ ./sdk/bin/jmonkeyplatform
Jun 23, 2022 4:32:29 PM org.netbeans.ProxyURLStreamHandlerFactory register
SEVERE: No way to find original stream handler for jar protocol
java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.net.URLStreamHandler java.net.URL.handler accessible: module java.base does not "opens java.net" to unnamed module @38082d64
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:180)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:174)
	at org.netbeans.ProxyURLStreamHandlerFactory.register(ProxyURLStreamHandlerFactory.java:59)
	at org.netbeans.JarClassLoader.<clinit>(JarClassLoader.java:117)
	at org.netbeans.MainImpl.execute(MainImpl.java:153)
	at org.netbeans.MainImpl.main(MainImpl.java:60)
	at org.netbeans.Main.main(Main.java:58)

java.lang.Error: factory already defined
	at java.base/java.net.URL.setURLStreamHandlerFactory(URL.java:1228)
	at org.eclipse.osgi.framework.internal.core.Framework.installURLStreamHandlerFactory(Framework.java:1765)
	at org.eclipse.osgi.framework.internal.core.Framework.<init>(Framework.java:232)
	at org.eclipse.osgi.framework.internal.core.EquinoxLauncher.internalInit(EquinoxLauncher.java:67)
	at org.eclipse.osgi.framework.internal.core.EquinoxLauncher.init(EquinoxLauncher.java:37)
	at org.eclipse.osgi.launch.Equinox.init(Equinox.java:178)
	at org.netbeans.modules.netbinox.Netbinox.init(Netbinox.java:63)
	at org.netbeans.core.netigso.Netigso.prepare(Netigso.java:145)
	at org.netbeans.NetigsoHandle.turnOn(NetigsoHandle.java:115)
	at org.netbeans.ModuleManager.enable(ModuleManager.java:1437)
	at org.netbeans.ModuleManager.enable(ModuleManager.java:1254)
	at org.netbeans.core.startup.ModuleList.installNew(ModuleList.java:315)
	at org.netbeans.core.startup.ModuleList.trigger(ModuleList.java:251)
	at org.netbeans.core.startup.ModuleSystem.restore(ModuleSystem.java:298)
	at org.netbeans.core.startup.Main.getModuleSystem(Main.java:156)
	at org.netbeans.core.startup.Main.getModuleSystem(Main.java:125)
	at org.netbeans.core.startup.Main.start(Main.java:282)
	at org.netbeans.core.startup.TopThreadGroup.run(TopThreadGroup.java:98)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

With `java-11-openjdk`:
```
$ ./sdk/bin/jmonkeyplatform
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.netbeans.ProxyURLStreamHandlerFactory (file:/opt/jme/sdk/platform/lib/boot.jar) to field java.net.URL.handler
WARNING: Please consider reporting this to the maintainers of org.netbeans.ProxyURLStreamHandlerFactory
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Inconsistency detected by ld.so: dl-lookup.c: 106: check_match: Assertion `version->filename == NULL || ! _dl_name_match_p (version->filename, map)' failed!
```

This makes it rather hard to use on Arch Linux for example, as java 10 doesn't come in the official repositories.

**Comment by tonihele** [comment]
Thanks for the report. The last one seems very familiar error but can't pin point now.

SDK 3.3 even comes with Java 11 (openjdk flavor) on which it runs on. So the installation went fine but just can't run it?

**Comment by vasilii-smirnov-globallogic** [comment]
Oh, just to be clear, I didn't try `jmonkeyplatform-linux-x64.sh`. I simply extracted `jmonkeyplatform.zip` 3.3, and called `bin/jmonkeyplatform`, which I assume uses the host jre. I don't really see anything that looks like openjdk in that zip file.

I also tried both openjdk and oracle jdk. The only one that ended up working for me was `java-10-openjdk`.

There is also this [AUR package](https://aur.archlinux.org/packages/jmonkeyplatform) of jmonkeyplatform that I tried, but it also uses the zip release and suffers the same problem.

Anyway, I'll try the sh installer in a few minutes and report how that goes...

**Comment by ea7ababe** [comment]
Yeah, ok, the one installed via the sh script runs just fine. It didn't even occur to me to try that. :)
Still it's a bit weird that the host jre can't run it. The versions are slightly different though:
```
$ java --version
openjdk 11.0.15 2022-04-19
OpenJDK Runtime Environment (build 11.0.15+10)
OpenJDK 64-Bit Server VM (build 11.0.15+10, mixed mode)
$ ./jmonkeyplatform/jdk/bin/java --version
openjdk 11.0.6 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)
```

**Comment by MeFisto94** [comment]
So to Java 17/18: I guess this works as soon as we upgrade to Netbeans 13/14, the SDK 3.3 release is too old for that.
For Java 11, we explicitly bundle adoptopenjdk / temurin, because of a nasty bug with libc/lwjgl2 that we can't get around easily.

The SDK cannot upgrade to lwjgl3 because of the Canvas/AWT Panels mode, but lwjgl2 is linked against libawt.so requiring `SUNW_private1.1` (or something) as a symbol, which is dropped after/with java 8 and then it depends on the libc version if it either a) ignores the symbol, b) crashes with the above assertion or c) (I think most recent ones do that) fail at loading the native library with a more meaningful message.

I tried to get lwjgl2 to build against a more modern JDK at https://github.com/MeFisto94/lwjgl but didn't pursue it further since our bundled AdoptOpenJDK seemed to work very well.

TLDR: If you want to see what's different, check the libc version used to build the JDK/try to inspect the symbols on libawt.so (I don't perfectly remember the specifics, but the engine repository certainly still has an issue created by me that goes very in depth)

**Comment by tonihele** [comment]
I'm closing this as a resolved issue. Please re-open if there is any further points. It'll be fixed in newer release and also the installation is very valid workaround for this.
